models/public/mozilla-deepspeech-0.8.2/pbmm_to_pb.py

```diff
@@ -0,0 +1,105 @@
+#! /usr/bin/env python3
+#
+# Copyright (C) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+import argparse
+
+try:
+    from tensorflow import GraphDef
+except ImportError:
+    from tensorflow.compat.v1 import GraphDef
+
+from mds_convert_utils.memmapped_file_system_pb2 import MemmappedFileSystemDirectory
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Convert memory-mapped Tensorflow protobuf format into plain Tensorflow protobuf (GraphDef)")
+    parser.add_argument('input', type=str, help="Input .pbmm filename")
+    parser.add_argument('output', type=str, help="Output .pb filename")
+    parser.add_argument('--overwrite', action='store_true', help="Overwrite output file if exists")
+    return parser.parse_args()
+
+
+def bytes_to_uint(data):
+    res = 0
+    for digit in reversed(data):
+        res = (res << 8) + digit
+    return res
+
+
+def load_memmapped_fs(data):
+    if isinstance(data, str):
+        # "data" is a filename, open it and read the data
+        with open(data, 'rb') as f:
+            data = f.read()
+
+    data_directory_ofs = bytes_to_uint(data[-8:])
+    data_directory = data[data_directory_ofs:-8]
+
+    directory_pb = MemmappedFileSystemDirectory()
+    directory_pb.ParseFromString(data_directory)
+
+    return {
+        entry.name: data[entry.offset : entry.offset + entry.length]
+        for entry in directory_pb.element
+    }
+
+
+def load_graph_def(data):
+    graph_def = GraphDef()
+    graph_def.ParseFromString(data)
+    return graph_def
+
+
+def serialize_graph_def(graph_def):
+    return graph_def.SerializeToString()
```

Comment:
This just seems like a useless function.

Reply:
Done: removed function, and removed load_graph_def as well, which contained only two lines.

This was done to make the top-level code readable. To my view, all these protobuf methods like "SerializeToString" etc. are incomprehensible.